src/Symfony/Bridge/PhpAmqp/Connection.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpAmqp;
+
+/**
+ * An AMQP connection.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class Connection
+{
+    private $amqpConnectionCredentials;
+    private $exchangeName;
+    private $queueName;
+    private $debug;
+
+    /**
+     * @var \AMQPChannel|null
+     */
+    private $amqpChannel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $amqpExchange;
+
+    /**
+     * @var \AMQPQueue|null
+     */
+    private $amqpQueue;
+
+    public function __construct(array $amqpConnectionCredentials, string $exchangeName, string $queueName, bool $debug = false)
```

Comment:
Given how we use classes for everything in this component (and it's great!) what if we replace this PHP array to store credentials by a proper `AMQPCredentials` class?

Reply:
As we are directly passing it to the `\AMQPConnection` object, I believe that it's better not to enforce any constraint on what can be in these "credentials". IMHO there is no point of maintaining an object mapping all the options available for the `\AMQPConnection` constructor coming from PHP's AMQP library.